ercc/main.go

```diff
@@ -34,39 +34,42 @@ func main() {
 		log.Panicf("error create enclave registry chaincode: %s", err)
 	}
 
-	// start chaincode as a service
-	config := serverConfig{
-		CCID:    os.Getenv("CHAINCODE_PKG_ID"),
-		Address: os.Getenv("CHAINCODE_SERVER_ADDRESS"),
-	}
+	ccid := os.Getenv("CHAINCODE_PKG_ID")
+	addr := os.Getenv("CHAINCODE_SERVER_ADDRESS")
 
-	server := &shim.ChaincodeServer{
-		CCID:    config.CCID,
-		Address: config.Address,
-		CC:      ercc,
-		TLSProps: shim.TLSProperties{
-			Disabled: true,
-		},
-	}
+	if len(ccid) > 0 && len(addr) > 0 {
+		// start chaincode as a service
+		config := serverConfig{
+			CCID:    ccid,
+			Address: addr,
+		}
+
+		server := &shim.ChaincodeServer{
+			CCID:    config.CCID,
+			Address: config.Address,
+			CC:      ercc,
+			TLSProps: shim.TLSProperties{
+				Disabled: true,
+			},
+		}
 
-	log.Printf("starting enclave registry (%s)\n", config.CCID)
+		log.Printf("starting enclave registry (%s)\n", config.CCID)
 
-	if err := server.Start(); err != nil {
-		log.Panicf("error starting enclave registry chaincode: %s", err)
+		if err := server.Start(); err != nil {
+			log.Panicf("error starting enclave registry chaincode: %s", err)
+		}
 	}
 
-	// alternatively we can start the chaincode in the normal way and let it connect the its peer
-	// TODO integrate the code below
-	// some switch is needed (e.g., via --arg or ENV_VAR) to start the chaincode in one or the other mode
-	// note that this code will also be shared with ecc
-
-	//ercc, err := contractapi.NewChaincode(&registry.Contract{})
-	//if err != nil {
-	//	log.Panicf("Error creating registry chaincode: %v", err)
-	//}
-	//
-	//if err := ercc.Start(); err != nil {
-	//	log.Panicf("Error starting registry chaincode: %v", err)
-	//}
+	if len(ccid) == 0 && len(addr) == 0 {
+		// alternatively we can start the chaincode in the normal way and let it connect the its peer
+		// TODO integrate the code below
+		// some switch is needed (e.g., via --arg or ENV_VAR) to start the chaincode in one or the other mode
+		// note that this code will also be shared with ecc
```

Comment:
This is obsoleted by your change!
```suggestion
		// start the chaincode in the traditional way
```